internal/ignition/ignition_test.go

```diff
@@ -921,15 +921,29 @@ var _ = Describe("downloadManifest", func() {
 	It("writes the correct file", func() {
 		ctx := context.Background()
 		manifestName := fmt.Sprintf("%s/manifests/openshift/masters-chrony-configuration.yaml", cluster.ID)
-		mockS3Client.EXPECT().Download(ctx, manifestName).Return(io.NopCloser(strings.NewReader("chronyconf")), int64(10), nil)
+		mockS3Client.EXPECT().Download(ctx, manifestName).Return(io.NopCloser(strings.NewReader("content:entry")), int64(10), nil)
 		Expect(os.Mkdir(filepath.Join(workDir, "/openshift"), 0755)).To(Succeed())
 		Expect(os.Mkdir(filepath.Join(workDir, "/manifests"), 0755)).To(Succeed())
 
 		Expect(generator.downloadManifest(ctx, manifestName)).To(Succeed())
 
 		content, err := os.ReadFile(filepath.Join(workDir, "/openshift/masters-chrony-configuration.yaml"))
 		Expect(err).NotTo(HaveOccurred())
-		Expect(content).To(Equal([]byte("chronyconf")))
+		Expect(content).To(Equal([]byte("content:entry")))
+	})
+
+	It("If a file has empty content, it will not be written", func() {
+		ctx := context.Background()
+		manifestName := fmt.Sprintf("%s/manifests/openshift/masters-chrony-configuration.yaml", cluster.ID)
+		mockS3Client.EXPECT().Download(ctx, manifestName).Return(io.NopCloser(strings.NewReader("")), int64(0), nil)
+		Expect(os.Mkdir(filepath.Join(workDir, "/openshift"), 0755)).To(Succeed())
+		Expect(os.Mkdir(filepath.Join(workDir, "/manifests"), 0755)).To(Succeed())
+
+		Expect(generator.downloadManifest(ctx, manifestName)).To(Succeed())
+
+		_, err := os.Stat(filepath.Join(workDir, "/openshift/masters-chrony-configuration.yaml"))
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("no such file or directory"))
```

Comment:
I think the better way to do this is `errors.Is(err, fs.ErrNotExist)`.

String matching is generally not a good choice.